segmentstore/server/src/test/java/io/pravega/segmentstore/server/DataRecoveryTestUtils.java

```diff
@@ -0,0 +1,188 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server;
+
+import com.google.common.base.Charsets;
+import io.pravega.common.Exceptions;
+import io.pravega.common.util.ArrayView;
+import io.pravega.common.util.AsyncIterator;
+import io.pravega.common.util.ByteArraySegment;
+import io.pravega.segmentstore.contracts.SegmentProperties;
+import io.pravega.segmentstore.contracts.StreamSegmentNotExistsException;
+import io.pravega.segmentstore.contracts.tables.IteratorArgs;
+import io.pravega.segmentstore.contracts.tables.IteratorItem;
+import io.pravega.segmentstore.contracts.tables.TableKey;
+import io.pravega.segmentstore.server.containers.DebugStreamSegmentContainer;
+import io.pravega.segmentstore.server.tables.ContainerTableExtension;
+import io.pravega.segmentstore.storage.SegmentHandle;
+import io.pravega.segmentstore.storage.Storage;
+import io.pravega.shared.segment.SegmentToContainerMapper;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+
+import static io.pravega.shared.NameUtils.getMetadataSegmentName;
+
+/**
+ * Utility methods for data recovery tests.
+ */
+@Slf4j
+public class DataRecoveryTestUtils {
+    private static final Duration TIMEOUT = Duration.ofSeconds(30);
+    private static final ScheduledExecutorService EXECUTOR_SERVICE = createExecutorService(10);
+
+    /**
+     * Lists all segments from a given long term storage.
+     * @param tier2             Long term storage.
+     * @param containerCount    Total number of segment containers.
+     * @return                  A map of lists containing segments by container Ids.
+     * @throws                  IOException in case of exception during the execution.
+     */
+    public static Map<Integer, List<SegmentProperties>> listAllSegments(Storage tier2, int containerCount) throws IOException {
+        SegmentToContainerMapper segToConMapper = new SegmentToContainerMapper(containerCount);
+        Map<Integer, List<SegmentProperties>> segmentToContainers = new HashMap<Integer, List<SegmentProperties>>();
+        log.info("Generating container files with the segments they own...");
```

Comment:
I don't see the point of this log message.

Reply:
Removed.